syntax = "proto3";
package peerservice.proto;



// peer service specific message:
message Heartbeat {
    bytes gmroot = 1; // highest known global merkle root
    int64 height = 2; // highest known ledger height
}


service PeerService {
    // any peer is able to request heartbeat from any other peer
    // it should send its own role: <merkle_root, height>
    // answer is peer's <merkle_root, height>
    rpc RequestHeartbeat (Heartbeat) returns (Heartbeat) {}
}